iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1896,8 +1896,6 @@ static void processDeviceMethodNotification(PMQTTTRANSPORT_HANDLE_DATA transport
                     transportData->transport_callbacks.method_complete_cb(STRING_c_str(method_name), payload->message, payload->length, (void*)dev_method_info, transportData->transport_ctx) != 0)
```

Comment:
So my recollection of this is that once we call the `method_complete_cb` then it takes ownership of the memory.  And it has to because of the async callback scenarios where we can't unconditionally free the memory in the `processDeviceMethodNotification`.  So this makes sense not to free it on an error case since we don't know how far things went for mem transfer.  (And looking down the callees are doing right things after your other fix.)

But... in the case that `mqttmessage_getApplicationMsg` on 1895 returns `NULL`, we still own freeing up the memory in this function since the callback sequence won't ever be called.  So we need to free in that case, correct?

It's also worth looking at the UT around this since I thought the failure conditions would've done a `mqttmessage_getApplicationMsg` fail and caught this leak?

Reply:
mqttmessage_getApplicationMsg is not mocked, so it never failed in the UT